From 88a1672fb803f735929414c03bee9767deeff9f7 Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Wed, 10 Aug 2011 02:13:21 +0200
Subject: [PATCH 1/3] Let TKService depend on X11 only when visualization
 component is compiled

This allows building OCAF and Data Exchange on computers without X11.

Closes issue #36.
---
 adm/cmake/TKService/CMakeLists.txt             |    2 +-
 inc/Aspect_XWD.hxx                             |    1 +
 src/AlienImage/AlienImage_X11XWDAlienData.cxx  |    4 +---
 src/AlienImage/AlienImage_X11XWDFileHeader.cxx |    4 +---
 4 files changed, 4 insertions(+), 7 deletions(-)

Index: oce/src/Aspect/Aspect_XWD.hxx
===================================================================
--- oce.orig/src/Aspect/Aspect_XWD.hxx
+++ oce/src/Aspect/Aspect_XWD.hxx
@@ -20,6 +20,7 @@
 
 # ifndef WNT
 #  include <X11/XWDFile.h>
+#  include <X11/X.h>
 # else
 
 # ifdef WORD64
Index: oce/src/AlienImage/AlienImage_X11XWDAlienData.cxx
===================================================================
--- oce.orig/src/AlienImage/AlienImage_X11XWDAlienData.cxx
+++ oce/src/AlienImage/AlienImage_X11XWDAlienData.cxx
@@ -29,9 +29,7 @@
 #define TEST	//GG_140699
 //		Check file extension, must be ".xwd".
 
-#if !defined(_WIN32) && !defined(__WIN32__) && (!defined(__APPLE__) || defined(MACOSX_USE_GLX))
-# include <X11/Xlib.h>
-#endif  // WNT
+#include <Aspect_XWD.hxx>
 #include <Aspect_GenericColorMap.hxx>
 #include <Image_PseudoColorImage.hxx>
 #include <AlienImage_MemoryOperations.hxx>
Index: oce/src/AlienImage/AlienImage_X11XWDFileHeader.cxx
===================================================================
--- oce.orig/src/AlienImage/AlienImage_X11XWDFileHeader.cxx
+++ oce/src/AlienImage/AlienImage_X11XWDFileHeader.cxx
@@ -16,9 +16,7 @@
 // purpose or non-infringement. Please see the License for the specific terms
 // and conditions governing the rights and limitations under the License.
 
-#if !defined(_WIN32) && !defined(__WIN32__) && (!defined(__APPLE__) || defined(MACOSX_USE_GLX))
-# include <X11/Xlib.h>
-#endif  // WNT
+#include <Aspect_XWD.hxx>
 #include <AlienImage_X11XWDFileHeader.hxx>
 
 const Handle(Standard_Type)& STANDARD_TYPE(AlienImage_X11XWDFileHeader)
